Fail to compile form source lxc on ubuntu:23.10

* Distribution:

Trying to compile in docker container, later to copy the results for ``make deps;...; make``  into target machine

``` bash
podman run --platform=linux/amd64 --name=lxd_ubuntu_23_10 -it ubuntu:23.10 bash
```

``` bash
root@f899528e51eb:/lxd# cat /etc/*release**
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=23.10
DISTRIB_CODENAME=mantic
DISTRIB_DESCRIPTION="Ubuntu 23.10"
PRETTY_NAME="Ubuntu 23.10"
NAME="Ubuntu"
VERSION_ID="23.10"
VERSION="23.10 (Mantic Minotaur)"
VERSION_CODENAME=mantic
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=mantic
LOGO=ubuntu-logo 
```

 * Distribution version:
Ubunut 23.10
main branch: https://github.com/canonical/lxd/commit/916dd99413236599fb2881602b4ff547f2dc7f44

 * The output of "lxc info" or if that fails:
``` bash
make deps
...
export CGO_CFLAGS="-I/root/go/deps/raft/include/ -I/root/go/deps/dqlite/include/"
export CGO_LDFLAGS="-L/root/go/deps/raft/.libs -L/root/go/deps/dqlite/.libs/"
export LD_LIBRARY_PATH="/root/go/deps/raft/.libs/:/root/go/deps/dqlite/.libs/"
export CGO_LDFLAGS_ALLOW="(-Wl,-wrap,pthread_create)|(-Wl,-z,now)"
root@f899528e51eb:/lxd#

FAIL
``` bash
root@f899528e51eb:/lxd# make
CGO_ENABLED=0 go install -v -tags lxd-metadata ./lxd/lxd-metadata
runtime/internal/atomic
runtime
internal/reflectlite
sync
internal/testlog
internal/bisect
errors
io
strconv
bytes
reflect
sort
internal/oserror
syscall
internal/syscall/unix
time
encoding/binary
encoding/base64
internal/fmtsort
internal/poll
internal/safefilepath
internal/syscall/execenv
path
io/fs
strings
os
bufio
context
fmt
hash
crypto
crypto/cipher
crypto/internal/boring
crypto/aes
encoding/json
crypto/internal/edwards25519/field
crypto/internal/nistec/fiat
embed
crypto/internal/randutil
crypto/internal/randutil: /usr/lib/go-1.21/pkg/tool/linux_amd64/compile: signal: segmentation fault (core dumped)
internal/godebug
math/rand
crypto/internal/nistec
math/big
crypto/sha512
crypto/sha256
crypto/des
crypto/elliptic
crypto/internal/bigmod
crypto/internal/boring/bbig
encoding/asn1
crypto/internal/edwards25519
crypto/hmac
vendor/golang.org/x/crypto/cryptobyte
crypto/md5
crypto/rc4
crypto/sha1
encoding/hex
crypto/x509/pkix
encoding/pem
vendor/golang.org/x/net/dns/dnsmessage
internal/singleflight
internal/intern
net/netip
net/url
net
path/filepath
DOMAIN=lxd
POFILES=$(wildcard po/*.po)
MOFILES=$(patsubst %.po,%.mo,$(POFILES))
LINGUAS=$(basename $(POFILES))
POTFILE=po/$(DOMAIN).pot
VERSION=$(or ${CUSTOM_VERSION},$(shell grep "var Version" shared/version/flex.go | cut -d'"' -f2))
ARCHIVE=lxd-$(VERSION).tar
HASH := \#
TAG_SQLITE3=$(shell printf "$(HASH)include <dqlite.h>\nvoid main(){dqlite_node_id n = 1;}" | $(CC) ${CGO_CFLAGS} -o /dev/null -xc - >/dev/null 2>&1 && echo "libsqlite3")
GOPATH ?= $(shell go env GOPATH)
CGO_LDFLAGS_ALLOW ?= (-Wl,-wrap,pthread_create)|(-Wl,-z,now)
SPHINXENV=doc/.sphinx/venv/bin/activate
SPHINXPIPPATH=doc/.sphinx/venv/bin/pip

ifneq "$(wildcard vendor)" ""
        RAFT_PATH=$(CURDIR)/vendor/raft
        DQLITE_PATH=$(CURDIR)/vendor/dqlite
else
        RAFT_PATH=$(GOPATH)/deps/raft
        DQLITE_PATH=$(GOPATH)/deps/dqlite
endif

        # raft
.PHONY: default
default: build

.PHONY: build
build:
ifeq "$(TAG_SQLITE3)" ""
        @echo "Missing dqlite, run \"make deps\" to setup."
        exit 1
endif

        CGO_ENABLED=0 go install -v -tags lxd-metadata ./lxd/lxd-metadata
        CC="$(CC)" CGO_LDFLAGS_ALLOW="$(CGO_LDFLAGS_ALLOW)" go install -v -tags "$(TAG_SQLITE3)" $(DEBUG) ./...
        CGO_ENABLED=0 go install -v -tags netgo ./lxd-migrate
        CGO_ENABLED=0 go install -v -tags agent,netgo ./lxd-agent
        @echo "LXD built successfully"

.PHONY: client
client:
        go install -v -tags "$(TAG_SQLITE3)" $(DEBUG) ./lxc
        @echo "LXD client built successfully"

.PHONY: lxd-agent
lxd-agent:
        CGO_ENABLED=0 go install -v -tags agent,netgo ./lxd-agent
        @echo "LXD agent built successfully"

.PHONY: lxd-migrate
lxd-migrate:
        CGO_ENABLED=0 go install -v -tags netgo ./lxd-migrate
        @echo "LXD-MIGRATE built successfully"
"Makefile" 308L, 9891B                                                                                                                                                                                                    1,1           Top
vendor/golang.org/x/crypto/chacha20
vendor/golang.org/x/crypto/internal/poly1305
io/ioutil
vendor/golang.org/x/sys/cpu
vendor/golang.org/x/crypto/chacha20poly1305
vendor/golang.org/x/crypto/hkdf
internal/saferio
compress/flate
encoding/gob
hash/crc32
compress/gzip
vendor/golang.org/x/text/transform
log
vendor/golang.org/x/text/unicode/norm
vendor/golang.org/x/text/unicode/bidi
vendor/golang.org/x/text/secure/bidirule
net/textproto
vendor/golang.org/x/net/idna
vendor/golang.org/x/net/http2/hpack
mime
vendor/golang.org/x/net/http/httpguts
vendor/golang.org/x/net/http/httpproxy
mime/quotedprintable
net/http/internal
net/http/internal/ascii
github.com/canonical/lxd/shared/ioprogress
github.com/canonical/lxd/shared/units
golang.org/x/sys/unix
database/sql/driver
github.com/kballard/go-shellquote
github.com/robfig/cron/v3
regexp/syntax
regexp
github.com/canonical/lxd/shared/osarch
gopkg.in/yaml.v2
os/exec
github.com/flosch/pongo2
github.com/pkg/xattr
os/user
encoding/csv
flag
text/template/parse
github.com/spf13/pflag
text/template
go/token
github.com/spf13/cobra
go/scanner
go/ast
go/build/constraint
go/internal/typeparams
go/parser
make: *** [Makefile:34: build] Error 1
```
  
# Issue description

Trying to compile in docker container, later to copy the results for ``make deps;...; make``  into target machine

# Steps to reproduce

 1. Step one
Run podman and container

``` bash
> podman machine stop
> podman machine set --cpus 2 --memory 4048
> podman machine start
This machine is currently configured in rootless mode. If your containers
require root permissions (e.g. ports < 1024), or if you run into compatibility
issues with non-podman clients, you can switch using the following command:

        podman machine set --rootful

API forwarding listening on: /var/run/docker.sock
Docker API clients default to this address. You do not need to set DOCKER_HOST.

Machine "podman-machine-default" started successfully
```  

 2. Step two
Start the container
```podman run --platform=linux/amd64 --name=lxd_ubuntu_23_10 -it ubuntu:23.10 bash```
 
3. Step three
Follow instruction from documentation https://documentation.ubuntu.com/lxd/en/latest/installing/
installing all os dependencies
```bash
apt update
apt install sudo acl attr autoconf automake dnsmasq-base git golang libacl1-dev libcap-dev liblxc1 liblxc-dev libsqlite3-dev libtool libudev-dev liblz4-dev libuv1-dev make pkg-config rsync squashfs-tools tar tcl xz-utils ebtables lvm2 thin-provisioning-tools btrfs-progs busybox-static curl gettext jq sqlite3 socat bind9-dnsutils vim -y
```

4. Step four (FAIL to compile)
Compile lxc
```
make deps # success

export CGO_CFLAGS="-I/root/go/deps/raft/include/ -I/root/go/deps/dqlite/include/"
export CGO_LDFLAGS="-L/root/go/deps/raft/.libs -L/root/go/deps/dqlite/.libs/"
export LD_LIBRARY_PATH="/root/go/deps/raft/.libs/:/root/go/deps/dqlite/.libs/"
export CGO_LDFLAGS_ALLOW="(-Wl,-wrap,pthread_create)|(-Wl,-z,now)"

make # fail
...
go/parser
make: *** [Makefile:34: build] Error 1
```

--- a/doc/installing.md
+++ b/doc/installing.md
@@ -171,7 +171,7 @@ sudo snap install --classic go
 If you use the `liblxc-dev` package and get compile time errors when building the `go-lxc` module,
 ensure that the value for `LXC_DEVEL` is `0` for your `liblxc` build. To check that, look at `/usr/include/lxc/version.h`.
 If the `LXC_DEVEL` value is `1`, replace it with `0` to work around the problem. It's a packaging bug, and
-we are aware of it for Ubuntu 22.04/22.10. Ubuntu 23.04/23.10 does not have this problem.
+we are aware of it for Ubuntu 22.04 onward, see [LP: #2039873](https://bugs.launchpad.net/ubuntu/+source/lxc/+bug/2039873).
 ```
 
 There are a few storage drivers for LXD besides the default `dir` driver.

